### PR TITLE
Update readme for xgboost-synthetic and remove outdated yaml file

### PR DESCRIPTION
**Pull request by abcdefgs0324**
Update README to let users execute the notebook in xgboost-synthetic example using one of the stock images launched through the UI.

/assign @jlewi 
/assign @zhenghuiwang

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/605)

### CODE REVIEW

File: xgboost_synthetic/README.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
-   ```
-1. Attach an extra data volume named 
+1. Open a notebook and run
+    ```
+    !git clone https://github.com/kubeflow/examples.git
+    ```
+
+1. In the directory `xgboost_synthetic`, open the notebook `build-train-deploy.ipynb`
+
+1. Set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` by the command
```
Comment (jlewi) [inline xgboost_synthetic/README.md]: Why is this needed? Isn't GOOGLE_APPLICATION_CREDENTIALS already set if they select add GCP credential in the notebook spawner UI?

File: xgboost_synthetic/build-train-deploy.ipynb [hunk 2]
```diff
@@ -29,6 +29,126 @@
     "Import the libraries required to train this model."
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
```
Comment (jlewi) [inline xgboost_synthetic/build-train-deploy.ipynb]: I don't think this is necessary per comment above. When you launch the notebook you should select "add gcp credential" and this should set GOOOGLE_APPLICATION_CREDENTIALS automatically

File: xgboost_synthetic/README.md (hunk 1 above)
Comment (abcdefgs0324) [reply xgboost_synthetic/README.md]: The section "Configurations" doesn't work when I launch a new notebook server using Kubeflow UI.
Another question is that there is no section to configure secrets or environment variables in the [Set up your notebook](https://www.kubeflow.org/docs/notebooks/setup/) page.

File: xgboost_synthetic/README.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+Kubeflow fairing, pipelines demo using synthetic data. This notebook `build-train-deploy.ipynb` can be executed using one of the stock notebook images launched through Kubeflow UI.
 
-1. Launch a notebook
+1. Follow the [Set up your notebook](https://www.kubeflow.org/docs/notebooks/setup/) guide to get started with Jupyter notebooks on Kubeflow
 
-   ```
-   kubectl apply -f notebook.xgboost-synthetic.yaml
-   ```
-1. Attach an extra data volume named 
+1. Open a notebook and run
```
Comment (zhenghuiwang) [inline xgboost_synthetic/README.md]: Nit: is it better to open a terminal and run `git` command?

Comment (abcdefgs0324) [reply xgboost_synthetic/README.md]: Fixed. Thanks
